### DISCUSSION TITLE
Send the favicon of the original URL instead of the favicon of YOURLS

### DISCUSSION THREAD

**Discussion by MAXOUXAX** [YOURLS/YOURLS#3418] [Q&A] [answered]
I searched for issues and plugins in the [awesome-yourls](https://github.com/YOURLS/awesome-yourls/) repository, and I couldn't find any plugins or solutions to make the favicon displayed the one from the original URL, rather than the one from YOURLS.

I ran into this problem when I added my short links to my YouTube channel links, which are displayed on my banner, since they all have the same icon, which makes them unintelligible :/

**Comment by dgw** [comment, marked as answer]
You would have to find some way for the server to determine what the target shorturl is, then look up the full URL associated with it and proxy that site's favicon instead—also keeping in mind that there are now [several different icons to serve up depending on who's looking](https://www.emergeinteractive.com/insights/detail/the-essentials-of-favicons/).

I'd guess that you don't see any plugins/solutions to do this because it is _hard_. When some random client—a YouTube server, say—requests `/favicon.ico` from yourdomain.tld, how is your server supposed to know for which shorturl (and therefore, which long URL) it's seeking an icon? Honestly, I think it's technically impossible to do this in the general case.

If you control the client(s), sure, it's possible to have the request include the info the server would need to return the long URL's favicon instead—which the server then needs to be set up to handle. But that solution won't apply to services like YouTube.
